exec/java-exec/src/main/java/org/apache/drill/exec/expr/stat/ParquetComparisonPredicate.java

```diff
@@ -26,29 +26,22 @@
 import java.util.ArrayList;
 import java.util.Iterator;
 import java.util.List;
-import java.util.function.BiPredicate;
 
 import static org.apache.drill.exec.expr.stat.ParquetPredicatesHelper.isNullOrEmpty;
 import static org.apache.drill.exec.expr.stat.ParquetPredicatesHelper.isAllNulls;
 
 /**
  * Comparison predicates for parquet filter pushdown.
  */
-public class ParquetComparisonPredicate<C extends Comparable<C>> extends LogicalExpressionBase
-    implements ParquetFilterPredicate<C> {
+public abstract class ParquetComparisonPredicate<C extends Comparable<C>> extends LogicalExpressionBase implements ParquetFilterPredicate<C> {
   private final LogicalExpression left;
+
   private final LogicalExpression right;
-  private final BiPredicate<Statistics<C>, Statistics<C>> predicate;
 
-  private ParquetComparisonPredicate(
-      LogicalExpression left,
-      LogicalExpression right,
-      BiPredicate<Statistics<C>, Statistics<C>> predicate
-  ) {
+  private ParquetComparisonPredicate(LogicalExpression left, LogicalExpression right) {
     super(left.getPosition());
     this.left = left;
     this.right = right;
-    this.predicate = predicate;
```

Comment:
What is/was wrong with the functional approach? Why is it necessary to use inheritance when delegation should work just fine here?

Reply:
The BiPredicate is perfect in case of boolean, which is not the case here. I return an enum which has no negate() for example...
Of course, I could create a new functional interface specifically for this enum, with only one function : test(), but it would only make the code more complicate.
I agree with you on this point, delegation is better than inheritance.

Reply:
There is no need to create a new functional interface. Use `BiFunction` (optionally define `RawMatchFunction<C>` as a concrete specification of the `BiFunction<Statistics<C>, Statistics<C>, ROWS_MATCH>`)

Reply:
Done